examples/misc/multiprocess.py

```diff
@@ -21,35 +22,33 @@ def __init__(self):
     def terminate(self):
         plt.close('all')
 
-    def poll_draw(self):
-
-        def call_back():
-            while 1:
-                if not self.pipe.poll():
-                    break
+    def call_back(self):
+        while True:
+            if not self.pipe.poll():
+                break
 
-                command = self.pipe.recv()
+            command = self.pipe.recv()
 
-                if command is None:
-                    self.terminate()
-                    return False
+            if command is None:
+                self.terminate()
+                return False
 
-                else:
-                    self.x.append(command[0])
-                    self.y.append(command[1])
-                    self.ax.plot(self.x, self.y, 'ro')
+            else:
+                self.x.append(command[0])
+                self.y.append(command[1])
+                self.ax.plot(self.x, self.y, 'ro')
 
-            self.fig.canvas.draw()
-            return True
-
-        return call_back
+        self.fig.canvas.draw()
```

Comment:
It is better to use `draw_idle` which defers the actually drawing until the GUI decides to really repaint the screen.